src/main/java/butler/TaskList.java

```diff
@@ -0,0 +1,193 @@
+package butler;
+
+import butler.task.Deadline;
+import butler.task.Event;
+import butler.task.Task;
+import butler.task.ToDo;
+
+import java.util.ArrayList;
+
+public class TaskList {
+
+    /** A list of Tasks */
+    protected ArrayList<Task> tasks;
+
+    /** Default constructor */
+    public TaskList() {
+        tasks = new ArrayList<>();
+    }
+
+    /**
+     * Gets the task from the Task ArrayList at the desired index.
+     *
+     * @param index Index of the Task of the ArrayList.
+     * @return Task at the desired index.
+     */
+    public Task get(int index) {
+        return tasks.get(index);
+    }
+
+    /**
+     * Gets the number of tasks in the TaskList.
+     *
+     * @return The size of the Task ArrayList, representing the number of tasks.
+     */
+    public int size() {
+        return tasks.size();
+    }
+
+    /**
+     * Finds all the tasks with descriptions that contain sub-strings that match the searchString.
+     *
+     * @param searchString String to match description sub-string.
+     * @return Butler response that contains all tasks with descriptions that contain sub-strings that
+     * match the searchString.
+     */
+    public String find(String searchString) {
+        int i = 0;
+        String output = "Here are the matching tasks in your list, Master: \n";
+        while (i < tasks.size()) {
+            Task task = tasks.get(i);
+            if (task.getDescription().contains(searchString)) {
+                output += String.valueOf(i + 1) + ". " + task.toString() + "\n";
+            }
+            i++;
+        }
+        return output;
+    }
+
+    /**
+     * Add a Deadline containing a description and a date and time to complete the task by a certain deadline.
+     *
+     * @param description description of the Task, which is a Deadline.
+     * @param dateTime String representing the date and time to complete the Deadline by.
+     * @return Butler response that confirms success status of adding the Deadline to the TaskList, and reiterates
+     * the task being added and the number of tasks currently in the list.
+     * @throws ButlerInputException Throws a ButlerInputException if user input is not satisfactory.
+     */
+    public String addDeadlineToList(String description, String dateTime) throws ButlerInputException {
+        if (description.equals("")) {
+            throw new ButlerInputException("Error. Sorry Master, but the description of a Deadline cannot be empty.\n");
```

Comment:
Hi Joey! I did it this way because of Single Responsibility Principle, having the message in the exception allows me to handle it and print the message should the exception be thrown. Having the Ui handle all the print statements is probably not the best idea if we have to refactor our code (to pass the message to a GUI for example)